### PR TITLE
[DIA-3017] Add the `UsNat` consent object in the callbacks

### PR DESCRIPTION
**Pull request by bohdan-go-wombat**
JIRA - [DIA-3017](https://sourcepoint.atlassian.net/browse/DIA-3017)

[DIA-3017]: https://sourcepoint.atlassian.net/browse/DIA-3017?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt [hunk 1]
```diff
@@ -367,50 +366,52 @@ private class ServiceImpl(
         consentActionImpl: ConsentActionImpl,
         env: Env,
         sPConsentsSuccess: ((SPConsents) -> Unit)?,
-        pmId: String?
+        pmId: String? // TODO figure out if we actually need this param, since we DO NOT use it
```
Comment (bohdan-go-wombat) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt:369]: @carmelo-iriti it seems like we do not use this param anywhere, do you think we can remove it?

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt [hunk 2]
```diff
@@ -367,50 +366,52 @@ private class ServiceImpl(
...
-                choiceType = consentActionImpl.actionType.toChoiceTypeParam(),
+                choiceType = consentAction.actionType.toChoiceTypeParam(),
                 accountId = spConfig.accountId.toLong(),
                 propertyId = spConfig.propertyId.toLong(),
                 env = env,
-                metadataArg = campaignManager.metaDataResp?.toMetaDataArg()?.copy(ccpa = null),
+                metadataArg = campaignManager.metaDataResp?.toMetaDataArg()?.copy(
+                    ccpa = null,
+                    usNat = null,
+                ),
```
Comment (bohdan-go-wombat) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt:414]: @carmelo-iriti in order to get choice for `GDPR` only, we have to null out meta data for other campaigns, i.e. `CCPA` and `UsNat`

Comment (carmelo-iriti) [reply cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt:414]: let me take a look I don’t remember this detail

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt (hunk 1 above)
Comment (bohdan-go-wombat) [reply cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt:369]: @carmelo-iriti I double checked, and neither the `pmId` from the `Service` interface, nor from the `ServiceImpl` is being used

`Service.kt`
image

`ServiceImpl.kt`
image

**Review by carmelo-iriti — APPROVED** [review]

**Pull request merged**